client/js/src/client.ts

```diff
@@ -73,8 +63,17 @@ export class Client {
 	}
 
 	eventSource_factory(url: URL): EventSource | null {
-		if (typeof window !== undefined && typeof EventSource !== "undefined") {
-			return new EventSource(url.toString());
+		if (typeof window === "undefined" || typeof EventSource === "undefined") {
+			import("eventsource")
+				.then((EventSourceModule) => {
+					global.EventSource = EventSourceModule.default as any;
+					return new EventSource(url.toString());
```

Comment:
We can just remove the assignment to the global namespace here, and called the module + return directly.